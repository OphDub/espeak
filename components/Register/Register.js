import React, { Component } from 'react';
import {
  StyleSheet,
  Image,
  Text,
  View,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import AwesomeAlert from 'react-native-awesome-alerts';
const logo = 'https://i.imgur.com/ana1fGy.png';

type Props = {};
export default class Register extends Component <Props>{
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      userName: '',
      password: '',
      confirmPassword: '',
      showAlert: false,
      alertMsg: '',
    };
  }

  validateRegistration = () => {
    const {
      email,
      userName,
      password,
      confirmPassword,
    } = this.state;
    const user = { email, userName, password };
    let alertMsg;

    if (email === '' || userName === '') {
      alertMsg = 'Please provide a email and username.';
      return this.setState({ showAlert: true, alertMsg });
    } else if (password === '' || confirmPassword === '') {
      alertMsg = 'Please add a password.';
      return this.setState({ showAlert: true, alertMsg });
    } else if (password !== confirmPassword) {
      alertMsg = 'Please make sure your passwords match.';
      return this.setState({ showAlert: true, alertMsg });
    }

    this.props.handleRegistration(user);
  }

  showAlert = () => {
    this.setState({ showAlert: true });
  }

  hideAlert = () => {
    this.setState({ showAlert: false });
  }

  render() {
    return (
      <View style={styles.container}>
        <Image
          style={styles.logo}
          source={{ uri: logo }}/>
        <Text style={styles.joinText}>Join eSpeak and learn Spanish!</Text>
        <TextInput
          style={styles.input}
          value={this.state.email}
          onChangeText={(email) => this.setState({ email })}
          placeholder="email"/>
        <TextInput
          style={styles.input}
          value={this.state.userName}
          onChangeText={(userName) => this.setState({ userName })}
          placeholder="your name"/>
        <TextInput
          style={styles.input}
          value={this.state.password}
          secureTextEntry={true}
          onChangeText={(password) => this.setState({ password })}
          placeholder="password"/>
        <TextInput
          style={styles.input}
          value={this.state.confirmPassword}
          secureTextEntry={true}
          onChangeText={(confirmPassword) => this.setState({ confirmPassword })}
          placeholder="confirm password"/>
        <TouchableOpacity
          style={styles.registerBtn}
          onPress={() => this.validateRegistration() }>
          <Text>Register</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => this.props.toggleRegistration() }
          style={styles.loginBtn}>
          <Text>
            Have an account? Login
          </Text>
        </TouchableOpacity>
        <AwesomeAlert
          show={this.state.showAlert}
          showProgress={false}
          title={'Uh oh! There is a problem with your registration!'}
          message={this.state.alertMsg}
          closeOnTouchOutside={true}
          closeOnHardwareBackPress={false}
          showCancelButton={false}
          showConfirmButton={true}
          confirmText="OK"
          confirmButtonColor="#3AAFb9"
          onConfirmPressed={() => {
            this.hideAlert();
          }}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  logo: {
    width: 100,
    height: 100,
    margin: 10,
  },
  joinText: {
    margin: 10,
  },
  input: {
    backgroundColor: '#CFCCD0',
    margin: 10,
    paddingVertical: 10,
    paddingHorizontal: 8,
    padding: 10,
    borderRadius: 8,
    width: 200,
  },
  registerBtn: {
    display: 'flex',
    width: 200,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#3AAFB9',
    paddingVertical: 10,
    borderRadius: 8,
    margin: 10,
    shadowColor: '#979797',
    shadowOpacity: 0.7,
    shadowOffset: { width: 2, height: 2 },
    shadowRadius: 4,
  },
  loginBtn: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
    width: 200,
    height: 25,
  }
});