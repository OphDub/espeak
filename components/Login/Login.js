import React, { Component } from 'react';
import {
  StyleSheet,
  Image,
  Text,
  View,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import Register from '../Register/Register';
import PropTypes from 'prop-types';
const logo = 'https://i.imgur.com/ana1fGy.png';

export default class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      toggleRegister: false
    };
  }

  renderLogin = () => {
    return (
      <View style={styles.container}>
        <Text style={styles.logoText}>
          eSpeak
        </Text>
        <Image
          style={styles.logo}
          source={{ uri: logo }}/>
        <TextInput
          style={styles.input}
          value={this.state.email}
          onChangeText={(email) => this.setState({ email })}
          placeholder="email"/>
        <TextInput
          style={styles.input}
          value={this.state.password}
          secureTextEntry={true}
          onChangeText={(password) => this.setState({ password })}
          placeholder="password"/>
        <TouchableOpacity
          onPress={() => this.props.handleLogin(this.state.email, this.state.password)}
          style={styles.loginBtn}>
          <Text style={styles.loginBtnText}>
            LOGIN
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => this.toggleRegistration()}
          style={styles.registerBtn}>
          <Text>
            New? Register here
          </Text>
        </TouchableOpacity>
      </View>
    );
  }

  renderRegister = () => {
    return (
      <Register
        handleRegistration={this.props.handleRegistration}
        toggleRegistration={this.toggleRegistration}
        beRegistration={this.props.beRegistration}/>
    );
  }

  renderView = () => {
    const { toggleRegister } = this.state;
    return toggleRegister ? this.renderRegister() : this.renderLogin();
  }

  toggleRegistration = () => {
    const { toggleRegister } = this.state;

    if(!toggleRegister) {
      this.setState({ toggleRegister: true });
    } else {
      this.setState({ toggleRegister: false });
    }
  }

  render() {
    return (
      <View style={styles.container}>
        { this.renderView() }
      </View>
    );
  }
}

Login.propTypes = {
  handleRegistration: PropTypes.func,
  handleLogin: PropTypes.func,
  beRegistration: PropTypes.func
};

const styles = StyleSheet.create({
  logo: {
    width: 100,
    height: 100,
    margin: 10,
  },
  logoText: {
    fontSize: 20
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  input: {
    backgroundColor: '#CFCCD0',
    margin: 10,
    paddingVertical: 10,
    paddingHorizontal: 8,
    padding: 10,
    borderRadius: 8,
    width: 200,
  },
  loginBtn: {
    display: 'flex',
    width: 200,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#3AAFB9',
    paddingVertical: 10,
    borderRadius: 8,
    margin: 10,
    shadowColor: '#979797',
    shadowOpacity: 0.7,
    shadowOffset: { width: 2, height: 2 },
    shadowRadius: 4,
  },
  loginBtnText: {
    color: '#FFFFFF'
  },
  registerBtn: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
    width: 200,
    height: 25,
  }
});