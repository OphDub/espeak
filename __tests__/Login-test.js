import React from 'react';
import { shallow, configure } from 'enzyme';
import Login from '../components/Login/Login';
import Register from '../components/Register/Register';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

describe('LOGIN', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<Login/>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  describe('LOGIN METHODS', () => {
    it('toggleRegistration should change toggleRegistration in state', () => {
      const wrapper = renderer.create(<Login/>);
      const instance = wrapper.getInstance();

      instance.toggleRegistration();

      expect(instance.state.toggleRegister).toEqual(true);
    });

    it('renderRegister should return a Register component', () => {
      const mockFn = jest.fn();
      const mockToggleRegistration = jest.fn().mockImplementation(() => {});
      const wrapper = renderer.create(<Login handleRegistration={mockFn} beRegistration={mockFn}/>);
      const instance = wrapper.getInstance();
      expect(instance.renderRegister()).toEqual(<Register handleRegistration={mockFn} beRegistration={mockFn} toggleRegistration={mockToggleRegistration}/>);
    });
  });
});