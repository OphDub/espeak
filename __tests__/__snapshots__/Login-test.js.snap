// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOGIN renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Welcome! Sign In:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="email"
      style={
        Object {
          "backgroundColor": "grey",
          "height": 25,
          "margin": 10,
          "width": 150,
        }
      }
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="password"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "grey",
          "height": 25,
          "margin": 10,
          "width": 150,
        }
      }
      value=""
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "purple",
          "display": "flex",
          "height": 25,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Login
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": 25,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        New user? Register
      </Text>
    </View>
  </View>
</View>
`;
